### PR TITLE
Include default handler capabilities to set a default handler when no other has been run

### PR DESCRIPTION
**Pull request by dagoonline**
Currently limited to a RESPOND method type.

Related to https://github.com/jobandtalent/bot/issues/179

### CODE REVIEW

File: robot.go [hunk 1]
```diff
@@ -10,14 +10,15 @@ import (
...
 type Robot struct {
-	Name       string
-	Alias      string
-	Adapter    Adapter
-	Store      Store
-	handlers   []handler
-	Users      *UserMap
-	Auth       *Auth
-	signalChan chan os.Signal
+	Name           string
```
Comment (fertapric) [inline robot.go:13]: Is this alignment imposed by the formatter? It doesn't help when diffing.

Comment (dagoonline) [reply robot.go:13]: Yes, `go fmt` for the good and for the bad.

Comment (fertapric) [reply robot.go:13]: There were long discussions regarding this kind of alignment in the Elixir formatter, so I know the story.

File: handler.go [hunk 2]
```diff
@@ -11,6 +11,9 @@ var (
 	respondRegexp = fmt.Sprintf(`^(?:@?(?:%s|%s)[:,]?)\s+(?:(.+))`, Config.Alias, Config.Name)
 	// RespondRegexpTemplate expands the RespondRegexp
 	respondRegexpTemplate = fmt.Sprintf(`^(?:@?(?:%s|%s)[:,]?)\s+(?:${1})`, Config.Alias, Config.Name)
+
+	// ErrNotMatched error to flag the case when there has been no match in the current handler
+	ErrNotMatched = fmt.Errorf("Handler not Matched")
```
Comment (fertapric) [inline handler.go]:
```suggestion
	ErrNotMatched = fmt.Errorf("Handler not matched")
```

File: handler.go [hunk 3]
```diff
@@ -67,10 +70,16 @@ func (h *Handler) Handle(res *Response) error {
 		return h.Run(res)
 	// if we don't find a match, return
 	default:
-		return nil
+		return ErrNotMatched
 	}
 }
 
+// IsRespond func
```
Comment (fertapric) [inline handler.go]: Are these kind of redundant comments normal in Go? Or this is disabling some warnings or similar?

**Review by fertapric — APPROVED** [review]

File: handler.go (hunk 3 above)
Comment (dagoonline) [reply handler.go]: They warn if an exported function or variable (those that starts with capital letter) is not commented. I missed a comment here.

Comment (dagoonline) [reply handler.go]: Fixed, thanks for pointing it out.

**Review by vcastellm — APPROVED** [review]

**Pull request merged**